tests/test_gs/test_header.py

```diff
@@ -60,16 +53,30 @@ def test_format_headers_raw(gs_raw_buffers):
     # can parse the header to find out why its not working.
     # dont need to display the images in the test pythons, but maybe generate the .avi file from the binary input
 
-from mio.devices.gs.header import buffer_to_array
+
+def test_buffer_npix(gs_raw_buffers):
+    format = GSBufferHeaderFormat.from_id("gs-buffer-header")
+    config = GSDevConfig.from_id("MSUS")
+    frame_buffers = defaultdict(list)
+
+    for i, buffer in enumerate(gs_raw_buffers):
+        header, pixels = GSBufferHeader.from_buffer(buffer, header_fmt=format, config=config)
+        frame_buffers[header.frame_num].append(pixels)
+
+    # discard first and last which may be incomplete in the sample
+    frames = list(frame_buffers.keys())
+    del frame_buffers[frames[0]]
+    del frame_buffers[frames[-1]]
+
+    for frame_num, pixels in frame_buffers.items():
+        pixel_lengths = [len(p) for p in pixels]
+        assert pixel_lengths == config.buffer_npix
```

Comment:
this is the only thing that this PR is about really, the rest is just getting the code to run again